src/Numerics/LinearAlgebra/Complex/DenseMatrix.cs

```diff
@@ -598,11 +598,24 @@ protected override void DoMultiply(Vector<Complex> rightSide, Vector<Complex> re
                     denseRight.Count,
                     1,
                     denseResult.Values);
+                return;
             }
-            else
+
+            if (rightSide.Storage is SparseVectorStorage<Complex> sparseRight)
```

Comment:
This logic appears 3 times in this file, only the matrix indices are swapped and/or its element is conjugated. I would extract the core part in one function, and call it 3 times; each time changing only the logic which manipulates the matrix value